superset/views.py

```diff
@@ -2544,11 +2545,19 @@ def csv(self, client_id):
         if not self.database_access(query.database):
             flash(get_database_access_error_msg(query.database.database_name))
             return redirect('/')
-
-        sql = query.select_sql or query.sql
-        df = query.database.get_df(sql, query.schema)
-        # TODO(bkyryliuk): add compression=gzip for big files.
-        csv = df.to_csv(index=False, encoding='utf-8')
+        blob = None
+        if results_backend and query.results_key:
+            blob = results_backend.get(query.results_key)
+        if blob:
+            json_payload = zlib.decompress(blob)
+            obj = json.loads(json_payload)
+            df = pd.DataFrame.from_records(obj['data'])
+            csv = df.to_csv(index=False, encoding='utf-8')
+        else:
+            sql = query.select_sql or query.executed_sql
```

Comment:
yes, query.executed_sql utilizes the db_engine_spec.PrestoEngineSpec.sql_preprocessor